RELEASE/scripts/autoscend/quests/level_11.ash

```diff
@@ -1000,6 +1009,18 @@ boolean L11_aridDesert()
 	{
 		return false;		//done exploring
 	}
+
+	if (auto_haveMaydayContract() && my_daycount() < 2 && have_effect($effect[Ultrahydrated]) == 0)
```

Comment:
I'm leaning towards removing check for ultrahydrated. Simply wait to d2 to get full benefit survival knife. Concern would be running out of stuff to do d1 I suppose

Add check for not having survival knife. Might have it D1 for some class/path combos

Reply:
Also feels like checking if survival knife is valid belongs here instead of `auto_haveMaydayContract()`

Reply:
I put checking if the survival knife is valid as I assume that will handle things correctly when Mayday rotates out of Standard as Mayday itself has no actual item or skill to test for that.

Reply:
> I'm leaning towards removing check for ultrahydrated. Simply wait to d2 to get full benefit survival knife. Concern would be running out of stuff to do d1 I suppose

Running out of stuff to do isn't a problem. We handle that in other places where we block quests for things like delay or shen snakes so we can handle this similarly.
I put the Ultrahydrated check in to not waste day 1 turns of Ultrahydrated but you're probably right it's best to just delay finishing until day 2 if we can save turns as we have stuff to do day 2 anyway and will get free turns of Ultrahydrated from BoFa on day 2 same as day 1.

> Add check for not having survival knife. Might have it D1 for some class/path combos

The classes that get it day 1 don't get it day 2. I want to have actual mafia functions to handle this properly though rather than hardcoding it like this but until mafia exposes the appropriate functions this will have to do.

Reply:
> > Add check for not having survival knife. Might have it D1 for some class/path combos
> 
> The classes that get it day 1 don't get it day 2. I want to have actual mafia functions to handle this properly though rather than hardcoding it like this but until mafia exposes the appropriate functions this will have to do.

Oh got it. Ty for explaining. Took me a bit!

> > Also feels like checking if survival knife is valid belongs here instead of `auto_haveMaydayContract()`
> 
> I put checking if the survival knife is valid as I assume that will handle things correctly when Mayday rotates out of Standard as Mayday itself has no actual item or skill to test for that.

Good call, didn't think about it not being an actual item